lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/dao/subscription/DBOSubscriptionDAOImpl.java

```diff
@@ -262,29 +293,50 @@ public SubscriptionPagedResults getSubscriptionList(String subscriberId,
 			results.setResults(new ArrayList<Subscription>(0));
 			results.setTotalNumberOfResults(0L);
 		} else {
-			String query = getListQuery(objectType);
 			MapSqlParameterSource parameters = new MapSqlParameterSource("ids", ids);
 			parameters.addValue("subscriberId", subscriberId);
 			parameters.addValue("objectType", objectType.name());
-			List<Subscription> subscriptions = namedTemplate.query(query, parameters, ROW_MAPPER);
+			List<Subscription> subscriptions = namedTemplate.query(SQL_GET_SUB_LIST, parameters, ROW_MAPPER);
 			results.setResults(subscriptions);
 			results.setTotalNumberOfResults((long) subscriptions.size());
 		}
 		return results;
 	}
 
-	public String getListQuery(SubscriptionObjectType objectType) {
-		ValidateArgument.required(objectType, "objectType");
-		switch (objectType) {
-			case FORUM:
-				return SQL_GET_FORUM_SUB_LIST;
-			case THREAD:
-				return SQL_GET_THREAD_SUB_LIST;
-			case DATA_ACCESS_SUBMISSION_STATUS:
-				return SQL_GET_SUB_LIST;
-			default:
-				throw new RuntimeException("Unsupported type "+objectType.name());
+	@Override
+	public SubscriptionPagedResults listSubscriptionForThread(String subscriberId, List<String> ids) {
+		ValidateArgument.required(subscriberId, "subscriberId");
```

Comment:
Why does this return a paginated result when it takes a list of ID and does not include limit or offset?